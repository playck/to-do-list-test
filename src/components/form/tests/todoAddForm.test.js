import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TodoAddForm from "../todoAddForm";
import renderer from "react-test-renderer";

describe("TodoAddForm", () => {
  it("redners", () => {
    const component = renderer.create(
      <TodoAddForm handleAddTodo={jest.fn()} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("Form Submit", () => {
    let handleAddTodo;
    let input;
    let button;

    beforeEach(() => {
      handleAddTodo = jest.fn();
      render(<TodoAddForm handleAddTodo={handleAddTodo} />);
      input = screen.getByPlaceholderText("할 일은?");
      button = screen.getByText("등록");
    });

    it("calls AddTodo when button is clicked and input not empty", () => {
      userEvent.type(input, "New Todo");
      userEvent.click(button);

      expect(handleAddTodo).toHaveBeenCalledWith("New Todo");
    });

    it("does not call AddTodo when the todo is empty", () => {
      // userEvent.type(input, "");
      // Error: Expected key descriptor but found "" in ""
      // falsy를 허용하지 않음
      userEvent.clear(input);
      userEvent.click(button);

      expect(handleAddTodo).toHaveBeenCalledTimes(0);
    });
  });
});
