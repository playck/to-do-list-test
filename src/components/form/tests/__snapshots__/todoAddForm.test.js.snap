// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAddForm redners 1`] = `
<form
  className="add_form"
  onSubmit={[Function]}
>
  <input
    className="add_input"
    placeholder="할 일은?"
    type="text"
  />
  <button
    className="add_button"
    style={
      Object {
        "fontSize": "10px",
      }
    }
  >
    등록
  </button>
</form>
`;
